cachi2/core/package_managers/npm.py

```diff
@@ -502,6 +566,95 @@ def _get_npm_dependencies(
     return download_paths
 
 
+def replace_dependency(dependency_version: str) -> bool:
+    """Check if dependency must be updated in package(-lock).json.
+
+    package(-lock).json files require to replace dependency URLs for
+    empty string in git and https dependencies in V2+.
+    """
+    url = urlparse(dependency_version)
+    if url.scheme == "file":
+        return False
+    return not (url.scheme == "" and "/" not in dependency_version)
+
+
+def update_package_lock_with_local_paths(
+    download_paths: Dict[NormalizedUrl, RootedPath],
+    package_lock: PackageLock,
+) -> None:
+    """Replace packages resolved URLs with local paths.
+
+    Update package-lock.json file in a way it can be used in isolated environment (container)
+    without internet connection. All package resolved URLs will be replaced for
+    local paths to downloaded dependencies.
+
+    :param download_paths:
+    :param package_lock: PackageLock instance which holds package-lock.json content
+    """
+    for package in package_lock.packages:
+        # Remove integrity for git sources, their integrity checksum will change when
+        # constructing tar archive from cloned repository
+        if package.resolved_url:
+            url = _normalize_resolved_url(str(package.resolved_url))
+        else:
+            continue
+        if _classify_resolved_url(url) == "git":
+            if package.integrity:
+                package.integrity = ""
+
+        # Update all type of dependencies except file:
```

Comment:
Maybe it's a little better to call it "package" here, since we're using "dependency" in the lines beneath to refer to the "dependencies" key inside each "package". 

Suggestion on how to rewrite this:

```
# Replace the resolved_url of all packages, unless it's already a file url:
```

Reply:
sounds good, changed